sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppNavigation.kt

```diff
@@ -15,18 +17,27 @@ fun AppNavigation(
         navController = navController,
         startDestination = startDestination
     ) {
-        composable(AppDestination.Home) {
+        composable(destination = AppDestination.Home) {
             HomeScreen(
-                navigator = { destination -> navController.navigate(destination) }
+                navigator = { destination ->
+                    navController.navigate(destination, destination.parcel)
+                }
             )
         }
 
-        composable(AppDestination.Second) { backStackEntry ->
+        composable(destination = AppDestination.Second) { backStackEntry ->
             SecondScreen(
                 navigator = { destination -> navController.navigate(destination) },
                 id = backStackEntry.arguments?.getString(KeyId).orEmpty()
             )
         }
+
+        composable(destination = AppDestination.Third) {
+            ThirdScreen(
+                navigator = { destination -> navController.navigate(destination) },
+                model = navController.previousBackStackEntry?.savedStateHandle?.get<UiModel>(KeyModel) ?: UiModel()
```

Comment:
@kaungkhantsoe I think we can allow `nullable` for `parcel`. What If we have many variables in `UiModel`? That means we need to set the default value in each variable. 🤔 

```suggestion
                model = navController.previousBackStackEntry?.savedStateHandle?.get<UiModel>(KeyModel)
```

So, by this way we no need to set the default value for `id` in `UiModel` cc @hoangnguyen92dn @Tuubz

Reply:
And one more thing, @kaungkhantsoe, I think we should create a new object `ModelParcel` and use this object for passing arguments instead of `UiModel`, What do you think @hoangnguyen92dn @Tuubz

Reply:
@Wadeewee Do you mean a wrapper class, likely `Bundle`, `Extra` to pack the real data object inside? It would me cumbersome since we have to create a new wrapper and in the destination one more step to get it out from the wrapper. I don't think we would need it 😄 Generic type is more convenient.

Reply:
+1 for allowing `nullable` for `parcel`, it will be easier to know if something went wrong on the next screen if so 🙏

Reply:
> Do you mean a wrapper class, likely Bundle, Extra to pack the real data object inside? 

Oh, @hoangnguyen92dn, I didn't mean like this ☝️ and What's I mean 👇  :

```kotlin
// ---- Use for UI Leyer ---- //
data class UiModel(
    val id: String
)

private fun Model.toUiModel() = UiModel(id.toString())

fun List<Model>.toUiModels() = this.map { it.toUiModel() }
```

```kotlin
// ---- Use for passing argument---- //
@Parcelize
data class ModelParcel(
    val id: String
) : Parcelable

private fun UiModel.toModelParcel() = ModelParcel(id = id)
```

```kotlin
 composable(destination = AppDestination.Third) {
            ThirdScreen(
                navigator = { destination -> navController.navigate(destination) },
                model = navController.previousBackStackEntry?.savedStateHandle?.get<ModelParcel>(KeyModel)
            )
        }
```
Actually, we don't use a kind of `UiModel` object to passing `argument` to another screen that's why I suggest to create a new object with suffix `Parcel` cc @Tuubz

Reply:
@Wadeewee So it is the same object but with a different suffix name and parcelized?

Reply:
@Wadeewee Gotcha 👍 It's good to separate the UiModel and Data Model for passing to the destination. As I mentioned, the sample is supposed to demo how we pass the data between screens, creating a new `ModelParcel` could share unnecessary boilerplate code IMO. cc @Tuubz

Reply:
> So it is the same object but with a different suffix name and parcelized?

@kaungkhantsoe, As I said in this [comment](https://github.com/nimblehq/android-templates/pull/390#discussion_r1092751052)

> Actually, we don't use a kind of UiModel object to passing argument to another screen

As it's a kind of `UiModel`, it was created/mapped from the response to render the `UI` on that screen. That's why we create an object with the suffix `Parcel` instead, and we do this way actually in client projects because when we navigate to the next screen, we mostly have a new `UiModel`, let's say `UiModelDetail` For more ref: You can research in `DeeMoney` project.

> As I mentioned, the sample is supposed to demo how we pass the data between screens, creating a new ModelParcel could share unnecessary boilerplate code

And also, @hoangnguyen92dn said here ☝️As it's the sample is supposed to demo how we pass the data between screens. So, It's an option for you to update then, but tbh, I don't think it's unnecessary because we usually do it in the client project. cc @hoangnguyen92dn @Tuubz 🙏

Reply:
I think keeping it as `UiModel` is fine, since it's still being used to compose the UI 🚀